dash_table/Format.py

```diff
@@ -0,0 +1,286 @@
+import collections
+import inspect
+import sys
+
+
+def get_named_tuple(name, dict):
+    return collections.namedtuple(name, dict.keys())(*dict.values())
+
+
+Align = get_named_tuple('align', {
+    'default': '',
+    'left': '<',
+    'right': '>',
+    'center': '^',
+    "right_sign": '='
+})
+
+Group = get_named_tuple('group', {
+    'no': '',
+    'yes': ','
+})
+
+Padding = get_named_tuple('padding', {
+    'no': '',
+    'yes': '0'
+})
+
+Prefix = get_named_tuple('prefix', {
+    'yocto': 10**-24,
+    'zepto': 10**-21,
+    'atto': 10**-18,
+    'femto': 10**-15,
+    'pico': 10**-12,
+    'nano': 10**-9,
+    'micro': 10**-6,
+    'milli': 10**-3,
+    'none': None,
+    'kilo': 10**3,
+    'mega': 10**6,
+    'giga': 10**9,
+    'tera': 10**12,
+    'peta': 10**15,
+    'exa': 10**18,
+    'zetta': 10**21,
+    'yotta': 10**24
+})
+
+Scheme = get_named_tuple('scheme', {
+    'default': '',
+    'decimal': 'r',
+    'decimal_integer': 'd',
+    'decimal_or_exponent': 'g',
+    'decimal_si_prefix': 's',
+    'exponent': 'e',
+    'fixed': 'f',
+    'percentage': '%',
+    'percentage_rounded': 'p',
+    'binary': 'b',
+    'octal': 'o',
+    'lower_case_hex': 'x',
+    'upper_case_hex': 'X',
+    'unicode': 'c'
+})
+
+Sign = get_named_tuple('sign', {
+    'default': '',
+    'negative': '-',
+    'positive': '+',
+    'parantheses': '(',
+    'space': ' '
+})
+
+Symbol = get_named_tuple('symbol', {
+    'no': '',
+    'yes': '$',
+    'binary': '#b',
+    'octal': '#o',
+    'hex': '#x'
+})
+
+Trim = get_named_tuple('trim', {
+    'no': '',
+    'yes': '~'
+})
+
+
+def is_string_py2(value):
+    return isinstance(value, str) or isinstance(value, unicode)
+
+
+def is_string_py3(value):
+    return isinstance(value, str)
+
+
+def is_string(value):
+    return is_string_py3(value) if sys.version_info >= (3, 0) else is_string_py2(value)
+
+
+class Format():
+    def __init__(self, **kwargs):
+        self._locale = {}
+        self._nully = ''
+        self._prefix = Prefix.none
+        self._specifier = {
+            'align': Align.default,
+            'fill': '',
+            'group': Group.no,
+            'width': '',
+            'padding': Padding.no,
+            'precision': '',
+            'sign': Sign.default,
+            'symbol': Symbol.no,
+            'trim': Trim.no,
+            'type': Scheme.default
+        }
+
+        valid_methods = [m for m in dir(self.__class__) if m[0] != '_' and m != 'to_plotly_json']
+
+        for kw, val in kwargs.items():
+            if kw not in valid_methods:
+                raise Exception('{0} is not a format method. Expected one of'.format(kw), str(list(valid_methods)))
+
+            getattr(self, kw)(val)
+
+    # Specifier
+    def align(self, value):
+        if value not in Align:
+            raise Exception('expected value to be one of', str(list(Align)))
```

Comment:
🌴 can these be pulled out, and called with something like `self._validate(value, Align)`?
Also the exception type - possibly `TypeError` again, unless we want `TypeError` for non-strings and `ValueError` for invalid strings - but that distinction is probably not worth adding extra code for https://docs.python.org/3/library/exceptions.html#TypeError
> Passing arguments of the wrong type (e.g. passing a list when an int is expected) should result in a TypeError, but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a ValueError.